src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -1733,6 +1736,48 @@ public void testRelativizedFileExclusion() throws Exception {
                 tempFile.getName(), expected);
     }
 
+    @Nested
+    public class TestProcessPathCollection {
+
+        @Test
+        public void testFileExtensionsPath() throws Exception {
+            final DefaultConfiguration checkerConfig = new DefaultConfiguration("configuration");
+            checkerConfig.addProperty("charset", StandardCharsets.UTF_8.name());
+            checkerConfig.addProperty("cacheFile", createTempFile("junit").getAbsolutePath());
+
+            final Checker checker = new Checker();
+            checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+            checker.configure(checkerConfig);
+
+            final DebugAuditAdapter auditAdapter = new DebugAuditAdapter();
+            checker.addListener(auditAdapter);
+
+            final List<Path> files = new ArrayList<>();
+            files.add(Paths.get("file.pdf"));
+            files.add(Paths.get("file.java"));
+            checker.setFileExtensions("java", "xml", "properties");
+            checker.setCacheFile(createTempFile("junit").getAbsolutePath());
+            final int counter = checker.process(files);
```

Comment:
should be tested by  some existing test method with verifyWithInlineXmlConfig

deprecated method we can tested by way you made this test, as it will be removed in some future.

Reply:
now this is implemented all over as its used internally within this `verifyWithInlineXmlConfig` program flow triggered, imho.